synapse/lib/storm.py

```diff
@@ -1162,10 +1171,31 @@ def getVar(self, name, defv=None):
             self.vars[name] = item
             return item
 
+        if self.root is not None:
+            valu = self.root.getVar(name, defv=s_common.novalu)
+            if valu is not s_common.novalu:
+                return valu
+
         return defv
 
+    def _isRootScope(self, name):
+        if self.root is None:
+            return False
+        if not self.funcscope:
+            return True
+        return name in self.root.vars
```

Comment:
discussed issue. should be `return self.root._isRootScope(name)`. Consider writing a 3-tier unit test (e.g. nested for loop)